cloudconfig/cloudinit/interface.go

```diff
@@ -509,12 +513,30 @@ type SSHKeyType string
 
 // The constant SSH key types sent to cloudinit through the cloudconfig
 const (
-	RSAPrivate SSHKeyType = "rsa_private"
-	RSAPublic  SSHKeyType = "rsa_public"
-	DSAPrivate SSHKeyType = "dsa_private"
-	DSAPublic  SSHKeyType = "dsa_public"
+	RSAPrivate     SSHKeyType = "rsa_private"
+	RSAPublic      SSHKeyType = "rsa_public"
+	DSAPrivate     SSHKeyType = "dsa_private"
+	DSAPublic      SSHKeyType = "dsa_public"
+	ECDSAPrivate   SSHKeyType = "ecdsa_private"
+	ECDSAPublic    SSHKeyType = "ecdsa_public"
+	ED25519Private SSHKeyType = "ed25519_private"
+	ED25519Public  SSHKeyType = "ed25519_public"
 )
 
+func NamesForSSHKeyType(sshPublicKeyType string) (private SSHKeyType, public SSHKeyType, err error) {
```

Comment:
Rename sshPublicKeyType to use the word algorithm otherwise type becomes very overloaded.